### PR TITLE
CI: Update GitHub Actions for set-output deprecation

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

GitHub Actions has deprecated set-output. Replace usages of set-output in stdout with the new syntax to save the output to the new environment variable.

See:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### Motivation and Context

Don't want deprecation warnings hiding real issues.

### How Has This Been Tested?

It hasn't. This was written by following documentation. A CI run will be the test.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: CI/check-services.py [hunk 1]
```diff
@@ -323,7 +323,7 @@ def main():
             service_authors = dict()
 
         # set GitHub outputs
-        print(f'::set-output name=make_pr::true')
+        print(f'make_pr=true') >> $GITHUB_OUTPUT
```
Comment (derrod) [inline CI/check-services.py:326]: This is not valid python, annoyingly this change makes it more difficult here. You'd have to write everything to a file.

Perhaps adding a helper for that would work, e.g.
```
def set_output(name, value):
    try:
        with open(os.environ['GITHUB_OUTPUT'], 'a', newline='\n') as f:
            f.write(f'{name}={value}')
    except Exception as e:
        print(f'Writing to github output files failed: {e!r}')
```
And then replace these calls with `set_output('make_pr', 'true')` etc.

Comment (RytoEX) [reply CI/check-services.py]: Removed the file from this PR to make it less troublesome. Rodney will revisit this point sometime later.

**Review by derrod — APPROVED** [review]

**Review by PatTheMav — APPROVED** [review]
Looks like a straight-forward change to me.

**Comment by RytoEX** [comment]
Per GitHub documentation, adjusted PowerShell (pwsh / PowerShell Core) syntax to:
```
"{name}=value" >> $env:GITHUB_OUTPUT
```

Running CI without and with "Seeking Testers" to ensure all is well, then will merge.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -449,7 +449,7 @@ jobs:
           CI/windows/03_package_obs.ps1 -CombinedArchs -Package
 
           $ArtifactName = (Get-ChildItem -filter "obs-studio-*-windows-x86+x64.zip" -File).Name
-          Write-Output "::set-output name=filename::${ArtifactName}"
+          "filename=${ArtifactName}" >> $env:GITHUB_OUTPUT
```
Comment (derrod) [inline .github/workflows/main.yml:452]: I believe this does not add a newline character by default, you should probably end each string with `\n`

Comment (RytoEX) [reply .github/workflows/main.yml:452]: The documentation doesn't specify.
See: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions?tool=powershell#setting-an-output-parameter

**Pull request merged**